gitlabform/gitlabform/test/test_merge_request_approvers.py

```diff
@@ -0,0 +1,234 @@
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approvers:
+        - merge_requests_user1
+"""
+
+config__approvers_only_groups = """
+gitlab:
+  api_version: 4
+
+project_settings:
+  gitlabform_tests_group/merge_requests_project:
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approver_groups:
+        - gitlabform_tests_group_with_user1_and_user2
+"""
+
+config__approvers_only_groups_change = """
+gitlab:
+  api_version: 4
+
+project_settings:
+  gitlabform_tests_group/merge_requests_project:
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approver_groups:
+        - gitlabform_tests_group_with_user1_and_user2
+        - gitlabform_tests_group_with_user4
+"""
+
+config__approvers_both = """
+gitlab:
+  api_version: 4
+
+project_settings:
+  gitlabform_tests_group/merge_requests_project:
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approvers:
+        - merge_requests_user3
+      approver_groups:
+        - gitlabform_tests_group_with_user4
+"""
+
+config__approvers_both_groups_change = """
+gitlab:
+  api_version: 4
+
+project_settings:
+  group1/project1:
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approvers:
+        - merge_requests_user3
+      approver_groups:
+        - gitlabform_tests_group_with_user4
+        - gitlabform_tests_group_with_user1_and_user2
+"""
+
+config__approvers_both_users_change = """
+gitlab:
+  api_version: 4
+
+project_settings:
+  group1/project1:
+    merge_requests:
+      approvals:
+        approvals_before_merge: 1
+        reset_approvals_on_push: true
+        disable_overriding_approvers_per_merge_request: true
+      approvers:
+        - merge_requests_user2
+        - merge_requests_user3
+      approver_groups:
+        - gitlabform_tests_group_with_user4
+"""
+
+
+class TestMergeRequestApprovers:
+
+    def test__if_it_works__only_users(self, gitlab):
+        gf = GitLabForm(config_string=config__approvers_only_users, project_or_group=GROUP_AND_PROJECT_NAME)
+        gf.main()
+
+        approvers = gitlab.get_approvers(GROUP_AND_PROJECT_NAME)
+        assert len(approvers['approvers']) == 2
+
+    def test__if_change_works__only_users(self, gitlab):
+        gf = GitLabForm(config_string=config__approvers_only_users, project_or_group=GROUP_AND_PROJECT_NAME)
+        gf.main()
+
+        # approvers = 2
```

Comment:
maybe you can add assertion here instead of comment - `assert len(approvers['approvers']) == 2` - it can be a self-describing code and sanity check that it indeed changed.
The same for `test__if_change_works__only_groups`